iml-gui/crate/src/components/lock_indicator.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    components::{font_awesome, popover, tooltip, Placement},
+    generated::css_classes::C,
+};
+use im::HashSet;
+use iml_wire_types::{warp_drive::Locks, LockChange, LockType, ToCompositeId};
+use seed::{prelude::*, *};
+
+pub(crate) fn lock_indicator<T>(x: &dyn ToCompositeId, all_locks: &Locks) -> Node<T> {
+    match all_locks.get(&x.composite_id().to_string()) {
+        Some(locks) if !locks.is_empty() => indicator(locks),
+        _ => empty![],
+    }
+}
+
+fn indicator<T>(lcks: &HashSet<LockChange>) -> Node<T> {
+    let (rw, ro): (HashSet<&LockChange>, HashSet<&LockChange>) =
+        lcks.iter().partition(|l| l.lock_type == LockType::Write);
+
+    let mut tooltip = vec![];
+    let mut popup = vec![];
+
+    let rw_c = rw.len();
+    if rw_c > 0 {
+        if rw_c > 1 {
```

Comment:
Nothing wrong with this but it might be nice to have a `pluralize` function somewhere:
```
tooltip.push(pluralize("{} write {}", rw_c, "lock"));
```

Reply:
This should be something similar to ngettext.